cypress/support/commands.ts

```diff
@@ -27,3 +29,53 @@ Cypress.Commands.add(
     }
   }
 );
+
+Cypress.Commands.add(
+  "isContainedInViewport",
+  { prevSubject: true },
+  (subject) => {
+    // @ts-ignore - Cypress.state is not typed
+    const window = Cypress.$(cy.state("window"));
+    const bottom = window.height();
+    const right = window.width();
+    const rect = subject[0].getBoundingClientRect();
+
+    // All corners of the element must be in the viewport
+    expect(rect.top).not.to.be.greaterThan(bottom);
+    expect(rect.bottom).not.to.be.greaterThan(bottom);
+    expect(rect.left).not.to.be.greaterThan(right);
+    expect(rect.right).not.to.be.greaterThan(right);
+
+    return subject;
```

Comment:
How about something like this?

```
    const rect = subject[0].getBoundingClientRect();
    const width = Cypress.config().viewportWidth;
    const height = Cypress.config().viewportHeight;

    // All corners of the element must be in the viewport
    expect(rect.top).to.be.within(0, height);
    expect(rect.right).to.be.within(0, width);
    expect(rect.bottom).to.be.within(0, height);
    expect(rect.left).to.be.within(0, width);

    return subject;
```

The tests passed when I tried it, though I'm not entirely sure what the difference is. ([link](https://spruce.mongodb.com/task/parsley_ubuntu1604_large_e2e_test_patch_84f5549e43a931d909770ab50065720bb8be2c8c_631763291e2d17647d8a7d1f_22_09_06_15_11_44/logs?execution=0))

Reply:
I think this might have to do with how chrome is rendering the text instead of an issue with the test itself

Reply:
So the bug was actually caused by the scrollbar of the page taking up space on linux. Macos treats the scrollbar as an overlay while linux treats it as an element with a set width.